gnosis_vpn-root/src/routing/macos.rs

```diff
@@ -60,89 +66,57 @@ impl Routing for Firewall {
             .ok()
             .ok_or(Error::General("Failed to acquire lock on pfctl".into()))?;
 
+        let interface = pfctl::Interface::from(&device);
+        let gw_ip: std::net::IpAddr = gateway
+            .ok_or(Error::General("No gateway found".into()))?
+            .as_str()
+            .parse()
+            .map_err(|e| Error::General(format!("failed to convert gatewat to IpAddr: {e}")))?;
+        let gw = pfctl::Ip::from(gw_ip);
+
         // Enable the firewall, equivalent to the command "pfctl -e":
         pf.try_enable()?;
 
-        // // Add an anchor rule for packet filtering rules into PF. This will fail if it already exists,
-        // // use `try_add_anchor` to avoid that:
-        // let anchor_name = "testing-out-pfctl";
-        // pf.add_anchor(anchor_name, pfctl::AnchorKind::Filter).unwrap();
-
-        // // Create a packet filtering rule matching all packets on the "lo0" interface and allowing
-        // // them to pass:
-        // let rule = pfctl::FilterRuleBuilder::default()
-        //     .action(pfctl::FilterRuleAction::Pass)
-        //     .interface("lo0")
-        //     .build()
-        //     .unwrap();
-
-        // // Add the filterig rule to the anchor we just created.
-        // pf.add_rule(anchor_name, &rule).unwrap();
+        // Add an anchor rule for packet filtering rules into PF. This will fail if it already exists,
+        // use `try_add_anchor` to avoid that:
+        pf.add_anchor(Firewall::ANCHOR_NAME, pfctl::AnchorKind::Filter)?;
+
+        // Create a packet filtering rule matching all packets on the identified interface and allowing
+        // them to pass:
+        // RULE: `pass out quick route-to (en0 192.168.0.1) from any to any user gnosisvpn`
+        let rule = pfctl::FilterRuleBuilder::default()
+            .action(pfctl::FilterRuleAction::Pass)
+            .interface(&device)
+            .direction(pfctl::Direction::Out) // or Any
+            .quick(true)
+            .route(pfctl::Route::RouteTo(pfctl::PoolAddr::new(interface, gw)))
```

Comment:
Is it possible to use this kind of routing setup if there is no gateway?

Reply:
Not that I know off.